core/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -326,4 +322,8 @@ public Version indexVersionCreated() {
     public QueryParseContext parseContext() {
         return this.parseContext;
     }
+
+    public Client client() {
```

Comment:
ok for that purpose then sure that makes sense. I'll go ahead and add such a method. Thanks.